bundles/org.openhab.binding.generacmobilelink/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,109 @@
+		<label>MobileLink Generator</label>
+		<description>Generator</description>
+		<channels>
+			<channel id="connected" typeId="connected"/>
+			<channel id="greenLight" typeId="greenLight"/>
+			<channel id="yellowLight" typeId="yellowLight"/>
+			<channel id="redLight" typeId="redLight"/>
+			<channel id="blueLight" typeId="blueLight"/>
+			<channel id="statusDate" typeId="statusDate"/>
+			<channel id="status" typeId="status"/>
+			<channel id="currentAlarmDescription" typeId="currentAlarmDescription"/>
+			<channel id="runHours" typeId="runHours"/>
+			<channel id="exerciseHours" typeId="exerciseHours"/>
+			<channel id="fuelType" typeId="fuelType"/>
+			<channel id="fuelLevel" typeId="fuelLevel"/>
+			<channel id="batteryVoltage" typeId="batteryVoltage"/>
+			<channel id="serviceStatus" typeId="serviceStatus"/>
+		</channels>
+		<representation-property>generatorId</representation-property>
+		<config-description-ref uri="thing-type:generacmobilelink:generator"/>
+	</thing-type>
+
+	<channel-type id="connected">
+		<item-type>Switch</item-type>
+		<label>Connected</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="greenLight">
+		<item-type>Switch</item-type>
+		<label>Green Light Status</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="yellowLight">
+		<item-type>Switch</item-type>
+		<label>Yellow Light Status</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="redLight">
+		<item-type>Switch</item-type>
+		<label>Red Light Status</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="blueLight">
+		<item-type>Switch</item-type>
+		<label>Blue Light Status</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="statusDate">
+		<item-type>String</item-type>
+		<label>Last Status Date</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="status">
+		<item-type>String</item-type>
+		<label>Status</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="currentAlarmDescription">
+		<item-type>String</item-type>
+		<label>Current Alarm Description</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="runHours">
+		<item-type>Number:Time</item-type>
+		<label>Number of Hours Run</label>
+		<state readOnly="true" pattern="%d %unit%"/>
+	</channel-type>
+	<channel-type id="exerciseHours">
+		<item-type>Number:Time</item-type>
+		<label>Number of Hours Exercised</label>
+		<state readOnly="true" pattern="%d %unit%"/>
+	</channel-type>
+	<channel-type id="fuelType">
+		<item-type>Number</item-type>
+		<label>Fuel Type</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="fuelLevel">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Fuel Level</label>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryVoltage">
+		<item-type>String</item-type>
+		<label>Battery Voltage Status</label>
+		<state readOnly="true"/>
+	</channel-type>
```

Comment:
Is the voltage returned as a number? Or is it a status like "Charging"?

Reply:
It's a String, like "good".  Unfortunately some of the variations on the possible return values are unknown to me.  There is no public documentation on this API, and i could find no other projects on the internet who have tried an integration, so this was me truly reverse engineering the API.